### PR TITLE
Update LinOSS discretizations and real-valued implementation

### PR DESCRIPTION
**Pull request by KasraMazaheri**
This PR keeps the LinOSS update focused on the core library code.

It adds newer LinOSS discretization and damping options, AG/RT initialization, stability projection, optional input normalization, multi-head LinOSS support, and optional output projection for multi-head mode.

It also changes the LinOSS forward path to use real-valued intermediates instead of complex JAX arrays, while preserving equivalence with the existing complex reference behavior. This should make the implementation friendlier to lower precision dtypes and accelerator kernels.

### CODE REVIEW

File: src/discretax/sequence_mixers/linoss.py [hunk 1]
```diff
@@ -18,21 +18,105 @@
...
+        -step * A / S,
+        step * (1 - step * G) / S,
+        1 / S,
+        step / S,
+        step**2 / S,
+    )
+
+
+def _mat_ex(A, G, step):
+    return 1 - step * G, -step * A, step, 1 + A * 0, step, A * 0
```
Comment (benedict-armstrong) [inline src/discretax/sequence_mixers/linoss.py:51]: `jnp.ones_like(A)` instead of `1 + A * 0` and `jnp.zeros_like(A)` instead of `A * 0`. I would say is a bit easier to read.

**Review by benedict-armstrong — APPROVED** [review]

File: src/discretax/sequence_mixers/linoss.py [hunk 2]
```diff
@@ -65,168 +169,392 @@ def __init__(
...
+            A, G = _project_ag_oscillatory(
+                self.discretization, A_diag, G_diag, steps, self.projection_eps
+            )
         else:
-            raise NotImplementedError(f"Discretization {self.discretization} not implemented")
+            A, G = _project_ag_stability(
+                self.discretization, A_diag, G_diag, steps, self.projection_eps
+            )
+        mat_fn = MATRIX_FNS[self.discretization]
+        return _apply_linoss(mat_fn, A, G, B, x, steps, gamma_log)
```
[the remark this answers is not in the document]
Comment (phnazari) [reply src/discretax/sequence_mixers/linoss.py:326]: @KasraMazaheri have a look at this. Correct? If not, resolve.

**Review by phnazari — CHANGES_REQUESTED** [review]
Have a look at the remaining comments please
